tests/io/test_agents.py

```diff
@@ -194,3 +195,55 @@ def test_agents_list_fields(api):
         check(agent, 'status', str)
         check(agent, 'uuid', 'uuid')
     assert count == agents.total
+
+
+@pytest.mark.vcr()
+def test_agents_unlink_agent_id_typeerror(api):
+    '''
+    test to raise the exception when type of agent_id is not as defined
+    '''
+    with pytest.raises(TypeError) as type_error:
+        api.agents.unlink('nope')
+    assert len(type_error.value.args) == 1, "Test-case should raise only one validation error."
+
+    assert type_error.value.args[0] == "agent_id is of type str.  Expected int", \
+        "Invalid type validation error for agent_id parameter is not raised by test-case."
+
+
+@pytest.mark.vcr()
+def test_agents_unlink_multiple_agent_id_typeerror(api):
+    '''
+    test to raise the exception when types of multiple agent_ids are not as defined
+    '''
+    with pytest.raises(TypeError) as type_error:
+        api.agents.unlink('nope', 'test', scanner_id=11)
+    assert len(type_error.value.args) == 1, "Test-case should raise only one validation error."
+
+    assert type_error.value.args[0] == "agent_ids is of type str.  Expected int", \
+        "Invalid type validation error for agent_ids parameter is not raised by test-case."
+
+
+@pytest.mark.vcr()
+def test_agents_task_status_scanner_id_typeerror(api):
+    '''
+    test to raise the exception when type of scanner_id is not as defined
+    '''
+    with pytest.raises(TypeError) as type_error:
+        api.agents.task_status(scanner_id='nope')
+    assert len(type_error.value.args) == 1, "Test-case should raise only one validation error."
+
+    assert "task_status() missing 1 required positional argument: 'task_uuid'" in type_error.value.args[0], \
+        "Missing value of required scanner_id parameter error is not raised by test-case."
+
+
+@pytest.mark.vcr()
+def test_agents_task_status_scanner_id_task_uuid_typeerror(api):
+    '''
+    test to raise the exception when type of task_uuid is not as defined
+    '''
+    with pytest.raises(TypeError) as type_error:
+        api.agents.task_status(task_uuid=11, scanner_id=11)
+    assert len(type_error.value.args) == 1, "Test-case should raise only one validation error."
+
+    assert "task_uuid is of type int.  Expected str" in type_error.value.args[0], \
+        "Missing value of required scanner_id & task_uuid parameter error is not raised by test-case."
```

Comment:
@bhavin-te , please check if this validation is only for `task_uuid` or for both `scanner_id` and `task_uuid` fields. Also, update test-case accordingly.

Reply:
Addressed comment